test/datastore_test.py

```diff
@@ -13,28 +16,30 @@ def test_doi_to_url(self, pudl_datastore_fixture):
         fake_doi = f"10.5072/zenodo.{number}"
 
         assert pudl_datastore_fixture.doi_to_url(fake_doi) == \
-            f"https://sandbox.zenodo.org/api/deposit/depositions/{number}"
+            f"{pudl_datastore_fixture.api_root}/deposit/depositions/{number}"
+
+    def test_sandbox_vs_production_dois(self):
+        """Check that Sandbox/Production DOIs are formatted as expected."""
+        for doi in datastore.DOI["sandbox"].values():
+            assert re.match(r"^10\.5072/zenodo\.[0-9]{5,10}$", doi) is not None  # noqa: FS003
+        for doi in datastore.DOI["production"].values():
+            assert re.match(r"^10\.5281/zenodo\.[0-9]{5,10}$", doi) is not None  # noqa: FS003
```

Comment:
it seems like there would be less duplication here if there was an argument here like `ds_type` or something that was either "sandbox" or "production". Or even something like:
```
for doi in list(datastore.DOI["production"].values()) + list(datastore.DOI["production"].values()):
    assert re.match(r"^10\.5281/zenodo\.[0-9]{5,10}$", doi) is not None

Reply:
Well, as it is now, there are two different patterns (reflecting the sandbox `10.5072` vs. production `10.5281` DOI patterns) and it tests all the production DOIs and all the sandbox DOIs whenever the test is run -- to make sure there hasn't been any confusion about which ones go where, cut-and-paste errors, etc. in the `datastore` module. I thought about doing some kind of `zip(("sandbox", "production"), ("5072", "5281"))` thing to iterate over but it seemed a little opaque.